cxx-qt-gen/src/parser/qobject.rs

```diff
@@ -107,4 +132,28 @@ mod tests {
         });
         assert!(qobject.parse_impl_items(&item.items).is_err());
     }
+
+    #[test]
+    fn test_parse_struct_fields_valid() {
+        let mut qobject = ParsedQObject::default();
+        let mut item: ItemStruct = tokens_to_syn(quote! {
+            struct T {
+                #[property]
+                property: f64,
+
+                field: f64,
+            }
+        });
+        assert!(qobject.parse_struct_fields(&mut item.fields).is_ok());
+        assert_eq!(qobject.properties.len(), 1);
+    }
+
+    #[test]
+    fn test_parse_struct_fields_invalid() {
+        let mut qobject = ParsedQObject::default();
+        let mut item: ItemStruct = tokens_to_syn(quote! {
+            struct T(f64);
+        });
+        assert!(qobject.parse_struct_fields(&mut item.fields).is_err());
```

Comment:
For now we only support named fields in the struct otherwise it errors in the fields_to_named etc.

And we don't support unnamed fields in structs yet, as later once the new property system comes in you'll need to access the Rust fields by their getter/setter methods we generate when you are in a C++ context (eg via prop() / prop_mut() ) so I'm  not sure how this would work with unnamed fields.

So for now best to error but maybe something to look at supporting later.